### PR TITLE
[ENG-8074][docs] update EAS Build images docs

### PR DESCRIPTION
**Pull request by szdziedzic**
# Why

We added a new Xcode 14.3 image and changed what `default` and `latest` means for M workers.

# How

Update images docs

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by amandeepmittal — APPROVED** [review]
Looks good to me. Thanks for update 👍

**Comment by kbrandwijk** [comment]
Do we want to use a bit more generic wording than 'm-medium' here to prevent us from having to update this again very very soon?

File: docs/pages/build-reference/infrastructure.mdx [hunk 1]
```diff
@@ -193,7 +209,7 @@ iOS worker VMs run on Mac Mini 8.1 hosts in an isolated environment. Every build
 
 </Collapsible>
 
-#### `macos-monterey-12.6-xcode-14.1` (alias `default`)
+#### `macos-monterey-12.6-xcode-14.1` (`default` for workers with Intel silicon)
```
Comment (dsokal) [inline docs/pages/build-reference/infrastructure.mdx:212]:
```suggestion
#### `macos-monterey-12.6-xcode-14.1` (`default` for Intel builders)
```

File: docs/pages/build-reference/infrastructure.mdx [hunk 2]
```diff
@@ -177,7 +177,23 @@ iOS worker VMs run on Mac Mini 8.1 hosts in an isolated environment. Every build
 
 ### iOS server images
 
-#### `macos-monterey-12.6-xcode-14.2` (alias `latest`)
+#### `macos-ventura-13.3-xcode-14.3` (`latest` for workers with Apple silicon, only available for workers with Apple silicon)
```
Comment (dsokal) [inline docs/pages/build-reference/infrastructure.mdx:180]:
```suggestion
#### `macos-ventura-13.3-xcode-14.3` (`latest` for Apple silicon builders, only available for Apple silicon builders)
```

File: docs/pages/build-reference/infrastructure.mdx [hunk 3]
```diff
@@ -177,7 +177,23 @@ iOS worker VMs run on Mac Mini 8.1 hosts in an isolated environment. Every build
...
+- Yarn 1.22.17
+- pnpm 8.2.0
+- npm 8.1.2
+- fastlane 2.212.2
+- CocoaPods 1.12.0
+- Ruby 2.7
+
+</Collapsible>
+
+#### `macos-monterey-12.6-xcode-14.2` (`latest` for workers with Intel silicon, `default` for workers with Apple silicon)
```
Comment (dsokal) [inline docs/pages/build-reference/infrastructure.mdx:196]:
```suggestion
#### `macos-monterey-12.6-xcode-14.2` (`latest` for Intel builders, `default` for Apple silicon builders)
```

**Review by dsokal — APPROVED** [review]
LGTM! 
Thank you for changing "worker" -> "builder". I grepped our docs and it seems we use this term on other pages too. If that's not too much hassle, could you make the change everywhere? It doesn't have to be in this PR :)

**Comment by szdziedzic** [comment]
>Thank you for changing "worker" -> "builder". I grepped our docs and it seems we use this term on other pages too. If that's not too much hassle, could you make the change everywhere? It doesn't have to be in this PR :)

I would preferably do it in a separate PR. Although I have a couple of things with a higher priority that I want to accomplish at the moment, so maybe somebody from the docs team would like to take this task?

**Review by wkozyra95 — APPROVED** [review]

**Pull request merged**